src/charm.py

```diff
@@ -5,16 +5,60 @@
 """Flask Charm service."""
 
 import logging
+import typing
 
-from ops.charm import CharmBase
+from ops.charm import CharmBase, ConfigChangedEvent
 from ops.main import main
+from ops.model import ActiveStatus
 
 logger = logging.getLogger(__name__)
 
 
 class FlaskCharm(CharmBase):
     """Flask Charm service."""
 
+    def __init__(self, *args: typing.Any) -> None:
+        """Initialize the instance.
+
+        Args:
+            args: passthrough to CharmBase.
+        """
+        super().__init__(*args)
+        self.framework.observe(self.on.config_changed, self.config_service)
+
+    def config_service(self, event: ConfigChangedEvent) -> None:
+        """Configure the flask pebble service layer.
+
+        Args:
+            event: the config-changed event that trigger this callback function.
+        """
+        container = self.unit.get_container("flask-app")
+        if container.can_connect():
+            container.add_layer("flask-app", self.flask_layer(), combine=True)
+            container.replan()
+            self.unit.status = ActiveStatus()
+        else:
+            event.defer()
+
+    def flask_layer(self) -> dict:
+        """Generate the pebble layer definition for flask application.
+
+        Returns:
+            The pebble layer definition for flask application.
+        """
+        return {
+            "services": {
+                "flask-app": {
+                    "override": "replace",
+                    "summary": "Flask application service",
+                    "command": "python3 -m gunicorn --chdir /srv/flask/app app:app -b 0.0.0.0:8080",
```

Comment:
Since the tech spec will expect a configuration file, I suggest to specify `chdir`
I would also keep the default port and binding address unless it causes issues for the charm ?

Reply:
Yes, I will move these command line arguments into a configuration file in the next story.
The default binding address is localhost, which is not accessible outside the container.

Reply:
Okay, no problem